api/src/main/java/openconsensus/OpenConsensus.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019, OpenConsensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package openconsensus;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import openconsensus.trace.NoopTrace;
+import openconsensus.trace.Trace;
+import openconsensus.trace.Tracer;
+
+/**
+ * Registry provides a global access to observability objects.
+ *
+ * <p>Implementations classes has to be registered via {@link ServiceLoader} e.g. <code>
+ * META-INF/services/openconsensus.trace.Tracer</code> for {@link Tracer}.
+ */
+public class OpenConsensus {
+
+  private static final OpenConsensus INSTANCE = new OpenConsensus();
+
+  private volatile Tracer tracer = Trace.getTracer();
+
+  /**
+   * Returns an instance of a {@link Tracer}.
+   *
+   * @return registered tracer or {@link NoopTrace} singleton via {@link Trace#getTracer()}.
+   */
+  public static Tracer getTracer() {
+    return INSTANCE.tracer;
+  }
+
+  private OpenConsensus() {
+    ServiceLoader<Tracer> tracers = ServiceLoader.load(Tracer.class);
+    Iterator<Tracer> tracerIterator = tracers.iterator();
+    if (tracerIterator.hasNext()) {
```

Comment:
There may be more than 1 `openconsensus.trace.Tracer` available with SPI on the classpath. I can foresee a situation where a user would desire a specific tracer to be registered with `OpenConsensus`, but the other one would be the first element returned from `tracers.iterator().next()`.

If `OpenConsensus` does not provide a way to work around this situation, the only thing the user could do is to remove the other `Tracer` from the classpath. This, however, is not possible from within a running application, and it also may not be possible before application start. An example of this could be a situation where an application or agent packages multiple tracers, and the desired tracer is specified at app start via system property.

Reply:
I agree we should take into account this use case.

Reply:
Perhaps there should be an `enabled` method on the interface that would allow them to be individually configured and skipped if not enabled?

Reply:
One solution would be to move the SPI load code out of the constructor and into a separate method, which would be required to be called by the user directly. Or, like @tylerbenson is suggesting, perhaps we can expose a system property that would allow a user to disable the SPI auto-loading.

Reply:
There is a system property `io.opentelemetry.trace.TracerProvider` which can be used to specify the tracer impl by supplying FQCN.

@SevaSafris would that satisfy your requirement?

Reply:
> One solution would be to move the SPI load code out of the constructor and into a separate method,

We wanted to allow use OT as traditional logging framework.

`private static final Tracer tracer = OpenTelemetry.getTracer()`